rustls/tests/unbuffered.rs

```diff
@@ -1422,3 +1314,123 @@ fn server_receives_incorrect_first_handshake_message() {
         _ => panic!("unexpected alert sending state"),
     };
 }
+
+const TLS12_CLIENT_TRANSCRIPT: &[&str] = &[
```

Comment:
Need a gut-check here: would it be overkill to try and lose the `Ok(..)` part of each of these while tidying up? It's a small bit of noise IMO since there's no `Err(..)` instances. It feels like something we could apply mechanically at the call-sites with a helper fn, but also like inconsequential nit pickery. This is a nice tidy on its own :+1:

Reply:
So I think these come from here:

```rust
fn advance_server(
    conn: &mut UnbufferedConnectionCommon<ServerConnectionData>,
    buffers: &mut Buffers,
    actions: Actions,
    transcript: &mut Vec<String>,
) -> State {
    let UnbufferedStatus { discard, state } = conn.process_tls_records(buffers.incoming.filled());

    transcript.push(format!("{:?}", state));

    let state = match state.unwrap() {
        ConnectionState::ReadEarlyData(mut state) => {
            let mut records = vec![];
            let mut peeked_len = state.peek_len();
```

I think it would be an improvement to `unwrap()` before we transcribe. 👍

Reply:
Ahh yeah that seems like a better idea indeed. Thanks!